core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala

```diff
@@ -726,8 +725,8 @@ class RDDSuite extends FunSuite with SharedSparkContext {
 
     val repartitioned = data.repartitionAndSortWithinPartitions(partitioner)
     val partitions = repartitioned.glom().collect()
-    assert(partitions(0) === Seq((0, 5), (0, 8), (2, 6)))
-    assert(partitions(1) === Seq((1, 3), (3, 8), (3, 8)))
+    assert(partitions(0).toSet === Set((0, 5), (0, 8), (2, 6)))
+    assert(partitions(1).toSet === Set((1, 3), (3, 8), (3, 8)))
```

Comment:
The original code will return `(0, 5), (0, 8), (2, 6)` and this PR will return `(0, 8), (0, 5), (2, 6)`. I think they both follow the criteria of sort based shuffle (keep the key ordering). Maybe the change is not so straightforward, but it is correct from my understanding.